cmake/tribits/common_tpls/FindTPLNetcdf.cmake

```diff
@@ -44,98 +44,158 @@
 # in order to trigger the right behavior in the function
 # tribits_tpl_find_include_dirs_and_libraries().
 #
-
-set(REQUIRED_HEADERS netcdf.h)
-set(REQUIRED_LIBS_NAMES netcdf)
-
-if (TPL_ENABLE_MPI)
-  set(REQUIRED_LIBS_NAMES ${REQUIRED_LIBS_NAMES} pnetcdf)
+if (${CMAKE_VERSION} GREATER "3.13")
+     cmake_policy(SET CMP0074 NEW)
 endif()
 
-#
-# Second, search for Netcdf components (if allowed) using the standard
-# find_package(NetCDF ...).
-#
-tribits_tpl_allow_pre_find_package(Netcdf  Netcdf_ALLOW_PREFIND)
-if (Netcdf_ALLOW_PREFIND)
-
-  message("-- Using find_package(Netcdf ...) ...")
-
-  set(CMAKE_MODULE_PATH
-    "${CMAKE_MODULE_PATH}"
-    "${CMAKE_CURRENT_LIST_DIR}/find_modules"
-    "${CMAKE_CURRENT_LIST_DIR}/utils"
-     )
+set(Netcdf_ALLOW_MODERN FALSE CACHE BOOL "Allow finding Netcdf as a modern CMake config file with exported targets (and only this way)")
 
-  find_package(NetCDF)
+if (Netcdf_ALLOW_MODERN)
 
-  if (NetCDF_FOUND)
-    set(DOCSTR "List of semi-colon separated paths to look for the TPL Netcdf")
-    set(TPL_Netcdf_Enables_Netcdf4 ${NetCDF_NEEDS_HDF5} CACHE BOOL
+  set(minimum_modern_netCDF_version 4.7.4)
+  message("-- Netcdf_ALLOW_MODERN=${Netcdf_ALLOW_MODERN}")
+  message("-- Using find_package(netCDF ${minimum_modern_netCDF_version} CONFIG) ...")
+  find_package(netCDF ${minimum_modern_netCDF_version} CONFIG)
+  if (netCDF_FOUND)
+    message("-- Found netCDF_CONFIG=${netCDF_CONFIG}")
+    message("-- Generating Netcdf::all_libs and NetcdfConfig.cmake")
+    # instead of using tribits_extpkg_create_imported_all_libs_target_and_config_file,
+    # we will do what it does ourselves because we need to bring in two
+    # inner find packages (netCDF and hdf5) because netCDF does not properly
```

Comment:
This is a really unfortunate chuck of duplication and would be a terrible precedent to set.  Might we consider having the TriBITS Netcdf TPL depend on the TriBITS HDF5 TPL and then have the Netcdf TPL depend on the HDF5 TPL as I tried to outline in https://github.com/TriBITSPub/TriBITS/issues/533#issuecomment-1285905951?  I think that would be very compact and you could essentially accomplish what this large amount of duplicated code is doing.  Is that something you would be willing to consider?  I can add a commit to this branch that attempts that (but I don't have a way to test it with `Netcdf_ALLOW_MODERN=ON`).

Reply:
Yes, I'm fine with moving in that direction. If you'd be willing to add a commit to this branch that would be super helpful, and I could test the modern option on my end.